crates/goose/src/agents/truncate.rs

```diff
@@ -544,6 +548,75 @@ impl Agent for TruncateAgent {
         let capabilities = self.capabilities.lock().await;
         capabilities.provider()
     }
+
+    async fn create_gooseling(&self, mut messages: Vec<Message>) -> Result<Gooseling> {
+        // get the gooseling prompt
+        let mut capabilities = self.capabilities.lock().await;
+        let system_prompt = capabilities.get_system_prompt().await;
+        let gooseling_prompt = capabilities.get_gooseling_prompt().await;
+        let provider = capabilities.provider();
+        let tools = capabilities.get_prefixed_tools().await?;
+
+        messages.push(Message::user().with_text(gooseling_prompt));
+
+        let (result, _usage) = provider.complete(&system_prompt, &messages, &tools).await?;
+
+        let content = result.as_concat_text();
+
+        // Use split_once to get the content after "Instructions:".
```

Comment:
Understand this was there before, but I don't love it. Could we update the prompt/add a tool to make the response/parsing easier?